flutter-studio/src/io/flutter/project/FlutterSettingsStep.java

```diff
@@ -140,6 +141,14 @@ protected ObservableBool canGoForward() {
 
   @Override
   protected void onEntering() {
+    Project project = getModel().project().getValueOrNull();
+    FlutterSdk sdk = FlutterSdk.forPath(getModel().flutterSdk().get());
+    boolean enableAndroidX = project == null && sdk != null && sdk.getVersion().isAndroidxSupported();
+    if (myAndroidXLabel.isVisible() != enableAndroidX) {
+      myAndroidXLabel.setVisible(enableAndroidX);
+      myUseAndroidxCheckBox.setVisible(enableAndroidX);
+      myRootPanel.doLayout();
```

Comment:
is the call to doLayout really needed here? I would expect you could invalidate instead of that the previous calls in this method would trigger invalidation.

Reply:
Yes, it is needed. I tried invalidate() instead but that had no effect. If the AndroidX section isn't visible when the initial layout is done then no vertical space is allocated to it. Changing visibility doesn't force re-layout (I guess -- not a Swing expert). And if it is visible initially then there's a big gap when it becomes invisible.